src/driver/mod.rs

```diff
@@ -342,6 +346,14 @@ impl<E: Engine> Driver<E> {
     }
 }
 
+/// Retrieves the L1 start block number.
+/// If an overflow occurs during subtraction, the function returns the genesis block (represented by `epoch_number`).
+fn get_l1_start_block(epoch_number: u64, channel_timeout: u64) -> u64 {
+    epoch_number
+        .checked_sub(channel_timeout)
+        .unwrap_or(epoch_number)
```

Comment:
Is this the correct logic? Shouldn't we return an epoch value of 0 if it underflows? Similar to how you changed the prune function.